ruvnl_consumer_app/app.py

```diff
@@ -94,6 +94,13 @@ def fetch_data(data_url: str, retry_interval: int = 30) -> pd.DataFrame:
         time.sleep(retry_interval)
         retries += 1
 
+    # after all retries, if no success, raise an exception
+    if retries == max_retries:
+        error_message = f"""Failed to fetch data after {max_retries} retries. 
+        Last status code: {r.status_code if 'r' in locals() else 'No response received'}"""
+        log.error(error_message)
+        raise ConnectionError(error_message)
+
```

Comment:
This is good, I think we might need to remove line 87, so we don't trigger this.

Could you add a test, perhaps with a fake url in it?